rx-lite/index.d.ts

```diff
@@ -232,7 +232,7 @@ declare namespace Rx {
         withLatestFrom<TOther, TResult>(souces: (Observable<TOther> | IPromise<TOther>)[], resultSelector: (firstValue: T, ...otherValues: TOther[]) => TResult): Observable<TResult>;
         concat(...sources: (Observable<T> | IPromise<T>)[]): Observable<T>;
         concat(sources: (Observable<T> | IPromise<T>)[]): Observable<T>;
-        concatAll(): Observable<T>;
+        concatAll(): T;
```

Comment:
Yes but this method only works for observable sequences of observables. Thus if we have:

If concatAll returns T:
```ts
let  x : Observable<Observable<number>>; //T is Observable<number>
let concat = x.concatAll();
//concat should be Observable<number>, not Observable<Observable<number>>
//this is only posible if concatAll() returns T

concat.subscribe(x => {  console.log(x + 1); /* x is a number */ });
```

If concatAll returns Observable(T) instead of T:
```ts
let concat = x.concatAll(); //concat is Observable<Observable<number>>

concat.subscribe(x => {  
console.log(x + 1); 
/* type error, x is a number in runtime, but 
typescript detects x as Observable<number> instead */ 
});
```